### ISSUE TITLE
problem when detaching scrollbar element from the body

### ISSUE THREAD

**Issue by PierBover** [gromo/jquery.scrollbar#49]
So I have some HTML that I load with AJAX and then store in an object. When I attach the HTML to the body the first time I initialise the plugin:

    $html.scrollbar();

The problem is that the plugin stops working when I detach the HTML and attach it again to the body. I've tried initialising it again but it does not work either.

Is this a bug or am I missing something?

TIA

**Comment by gromo** [comment]
Hi @PierBover 

I cannot say what's wrong exactly, but I think you shouldn't do it that way. I think the problem is that you manipulate original container and forget about wrapper that is added by custom scrollbar. You can init scrollbar on some wrapper inside your body and update it's content with your html.

**Comment by PierBover** [comment]
@gromo That was my first approach, but then I needed to add a scroll event only for one of the scrollable contents (to detect the scrolling position).

**Comment by gromo** [comment]
@PierBover
it's still not clear for me what you're doing and what is the problem. Can you prepare simple (as simple as possible) example for me to look at?

**Comment by PierBover** [comment]
I was loading content dynamically into the scrollable area. For some content I needed to know the scroll position using a `scroll()` event to do some animations when something becomes visible for the user.

If I had a single scrollable div, and changed it's content dynamically, I would have to add and remove the event listener dynamically depending on the content to prevent the scroll event triggering forever.

So instead of having a scroll zone in the page, I added a scrollable div wrapper to each content. The problem was that your scroll plugin only worked the first time I attached it to the DOM. When I reattached the content to the DOM it stops working.

I needed to solve this so I ended up not using your plugin, but that was the problem.
